src/modes/insert.rs

```diff
@@ -157,6 +171,28 @@ impl Mode for Insert {
                     let delta = ops::delete_cursor(&buffer.data, &buffer.selection);
                     ModeTransition::DirtyBytes(buffer.apply_incomplete_delta(delta))
                 }
+                Action::Move(direction) => {
+                    // we should not move if user already write a half of the hex byte
+                    if self.hex_half.is_none() {
+                        let max_bytes = buffer.data.len();
+                        ModeTransition::new_mode_and_dirty(
+                            Insert {
+                                before: self.before,
+                                hex: self.hex,
+                                hex_half: self.hex_half,
+                            },
+                            buffer.map_selections(|region| {
+                                vec![region.simple_move(direction, bytes_per_line, max_bytes, 1)]
+                            }),
+                        )
+                    } else {
+                        ModeTransition::new_mode(Insert {
```

Comment:
This can be even further simplified to `ModeTransition::None` -- sorry about the lacking documentation!